source/distributed/tria.cc

```diff
@@ -1689,34 +2225,48 @@ namespace parallel
 
       if (this->my_subdomain == 0)
         {
-          const unsigned int buffer_size_per_cell =
-            (cell_attached_data.cumulative_fixed_data_size > 0 ?
-               cell_attached_data.cumulative_fixed_data_size +
-                 sizeof(CellStatus) :
-               0);
-
           std::string   fname = std::string(filename) + ".info";
           std::ofstream f(fname.c_str());
-          f << "version nproc attached_bytes n_attached_objs n_coarse_cells"
-            << std::endl
-            << 2 << " "
+          f << "version nproc n_attached_objs n_coarse_cells" << std::endl
+            << 3 << " "
             << Utilities::MPI::n_mpi_processes(this->mpi_communicator) << " "
-            << buffer_size_per_cell << " "
             << cell_attached_data.pack_callbacks.size() << " "
             << this->n_cells(0) << std::endl;
         }
 
-      if (cell_attached_data.cumulative_fixed_data_size > 0)
+#  ifdef DEBUG
+      // each cell should have been flagged `CELL_PERSIST`
+      for (auto &qcrel : local_quadrant_cell_relations)
         {
-          const_cast<
-            dealii::parallel::distributed::Triangulation<dim, spacedim> *>(this)
-            ->attach_mesh_data();
+          const auto &cell_status = std::get<1>(qcrel);
+          Assert(
+            (cell_status ==
+             parallel::distributed::Triangulation<dim, spacedim>::CELL_PERSIST),
+            ExcInternalError());
         }
+#  endif
 
-      dealii::internal::p4est::functions<dim>::save(
-        filename,
-        parallel_forest,
-        cell_attached_data.cumulative_fixed_data_size > 0);
+      if (cell_attached_data.n_attached_datas > 0)
```

Comment:
I had not noticed this before, but you have the same issue here with "datas" not being a real word.

Reply:
The person who introduced the error may in fact have been me. Feel free to ignore the issue if you want.